kong/plugins/rate-limiting/migrations/003_10_to_112.lua

```diff
@@ -0,0 +1,13 @@
+return {
+  postgres = {
+    up = [[
+      CREATE INDEX IF NOT EXISTS ratelimiting_metrics_idx ON ratelimiting_metrics (service_id, route_id, period_date, period);
+    ]],
+  },
+
+  cassandra = {
+    up = [[
+      CREATE INDEX IF NOT EXISTS ratelimiting_metrics_idx ON ratelimiting_metrics (service_id, route_id, period_date, period);
```

Comment:
Looks like CI is failing on Cassandra due to trying to create a composite index...
https://travis-ci.org/Kong/kong/jobs/518351188#L267
...and looks like this is not supported (sorry for not realizing it earlier!)
https://docs.datastax.com/en/cql/3.3/cql/cql_reference/cqlCreateIndex.html

One interesting thing is that, in spite of the performance degradation observed when migrating to 0.14 in #4374, as far as I can tell Kong 0.13 or earlier did _not_ have indices to these tables either.

What version did you migrate from, btw? Perhaps the degradation came from the Postgres migration that added route_id and service_id and recreated the primary key? (Cassandra migrations drop the table altogether and recreate it)

If the added index fixed your observed performance issues on Postgres, I guess we can add it for Postgres only then.

Reply:
The problem arose when we migrated from Kong 0.11 to kong 0.14 and we remove all APIs to Services + Routes.  We had Kong running for more than a year, and had more than a year of rate-limiting data in the table.  The problem was only observed on Postgres.